src/app/main/component/user/components/profile/profile-service/profile.service.ts

```diff
@@ -46,6 +47,32 @@ export class ProfileService {
     return this.http.get<FactOfTheDay>(`${mainLink}fact-of-the-day/random?lang=${currentLang}`);
   }
 
+  getHabitFactFromLocalStorage(): FactOfTheDay | null {
+    const savedHabitFact = localStorage.getItem('habitFactOfTheDay');
+    const lastHabitFetchTime = localStorage.getItem('lastHabitFactFetchTime');
+    const currentTime = Date.now();
+
+    if (savedHabitFact && lastHabitFetchTime && currentTime - Number(lastHabitFetchTime) < this.oneDayInMillis) {
+      return JSON.parse(savedHabitFact);
+    }
+    return null;
+  }
+
+  saveHabitFactToLocalStorage(fact: FactOfTheDay, currentTime: number): void {
```

Comment:
The main purpose of the `ProfileService` is to fetch data from the server
If you need `SaveHabitFactToLocalStorage` or `clearHabitFactFromLocalStorage`, you can create them in LocalStorageService